roles/sap_netweaver_preconfigure/tasks/sapnote/3119751.yml

```
@@ -5,23 +5,54 @@
     msg: "SAP note {{ (__sap_netweaver_preconfigure_sapnotes_versions | selectattr('number', 'match', '^3119751$') | first).number }}
           (version {{ (__sap_netweaver_preconfigure_sapnotes_versions | selectattr('number', 'match', '^3119751$') | first).version }}): Linux Requirements for SAP Kernel 754 and for SAP Kernel 788 and higher"
 
-- name: Get info about the compat-sap-c++-10.so file
-  ansible.builtin.stat:
-    path: /opt/rh/SAP/lib64/compat-sap-c++-10.so
-  register: __sap_netweaver_preconfigure_register_stat_compat_sap_cpp
+# Note: This file is only included for RHEL 8, so no further when condition is required here.
 
-- name: Create directory '{{ sap_netweaver_preconfigure_rpath }}'
+- name: Identify all 'compat-sap-c++-NUM.so' files with NUM >= 10
+  ansible.builtin.find:
+    paths: '/opt/rh/SAP/lib64'
+    patterns: "compat-sap-c++-1*"
+  register: __sap_netweaver_preconfigure_register_find_compat_sap_cpp
+
+# Note: The file compat-sap-c++-NUM.so with NUM >= 10 will be available if the role sap_general_preconfigure has been run before,
+# which is a requirement.
+- name: Fail if there is no file 'compat-sap-c++-NUM.so' file with NUM >= 10
+  ansible.builtin.fail:
+    msg: There is no file '/opt/rh/SAP/lib64/compat-sap-c++-NUM.so' file with NUM >= 10!
+  when: __sap_netweaver_preconfigure_register_find_compat_sap_cpp.matched == 0
+
+# Note: All following tasks depend on the previous task not having failed, so no further when condition is used below.
+- name: Set fact for file names from find result
+  ansible.builtin.set_fact:
+    __sap_netweaver_preconfigure_fact_compat_sap_cpp_filenames: "{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_filenames | d([]) + [__sap_netweaver_preconfigure_item.path | basename] }}"
+  loop: "{{ __sap_netweaver_preconfigure_register_find_compat_sap_cpp.files }}"
+  loop_control:
+    loop_var: __sap_netweaver_preconfigure_item
+    label: "{{ __sap_netweaver_preconfigure_item.path }}"
+
+- name: Set fact for the latest 'compat-sap-c++.NUM.so' file name
+  ansible.builtin.set_fact:
+    __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest: "{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_filenames | sort | last }}"
```

Another suggestion to simplify the processing of the file query results by using map filters and combining 2 tasks since it got rid of the step going through the loop.

The following example gets a list of the `path` elements, sorts and reduces to the last file and strips the path from that file.
The outcome is of type `AnsibleUnsafeText` and not unicode. No idea if that makes any difference in further processing, hence mentioning.

Tested example:
```
- name: find files
  find:
    path: /etc
  register: etc_files

- set_fact:
    lastfile: >
      {{ etc_files.files
        | map(attribute = 'path')
	| sort | last | basename }}

- debug:
    var: lastfile
```


```suggestion
- name: Set fact for the latest 'compat-sap-c++.NUM.so' file name
  ansible.builtin.set_fact:
    __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest: >
      {{ __sap_netweaver_preconfigure_register_find_compat_sap_cpp.files
        | map(attribute = 'path')
        | sort | last | basename }}
```

---

Thanks for the suggestion. The proposed code does not work but with a little modification, I could solve this problem. Here's my test playbook:
```
    - name: Set fact for the latest 'compat-sap-c++.NUM.so' file name
      ansible.builtin.set_fact:
        __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest_1: >
          {{ __sap_netweaver_preconfigure_register_find_compat_sap_cpp.files
            | map(attribute = 'path')
            | sort | last | basename }}

    - name: debug 1
      debug:
        msg:
          - ">>>{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest_1 }}<<<"

    - name: Set fact for the latest 'compat-sap-c++.NUM.so' file name
      ansible.builtin.set_fact:
        __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest_2: "{{ __sap_netweaver_preconfigure_register_find_compat_sap_cpp.files
                                                                   | map(attribute = 'path') | sort | last | basename }}"

    - name: debug 2
      debug:
        msg:
          - ">>>{{ __sap_netweaver_preconfigure_fact_compat_sap_cpp_latest_2 }}<<<"
```

Output:
```TASK [Set fact for the latest 'compat-sap-c++.NUM.so' file name] ***************************************************************************
ok: [localhost]

TASK [debug 1] *****************************************************************************************************************************
ok: [localhost] => 
  msg:
  - |-
    >>>compat-sap-c++-10.so
    <<<

TASK [Set fact for the latest 'compat-sap-c++.NUM.so' file name] ***************************************************************************
ok: [localhost]

TASK [debug 2] *****************************************************************************************************************************
ok: [localhost] => 
  msg:
  - '>>>compat-sap-c++-10.so<<<'
```

---

Yes, my mistake with the suggestion syntax due to the rewrite.
It works when changing `>` to `>-` to strip trailing whitespace:

```
__sap_netweaver_preconfigure_fact_compat_sap_cpp_latest_1: >-
```

---

@ja9fuchs Thanks for the suggestion!

---

Solved with the latest commits.